### PR TITLE
No keyring option and fixes

### PR DESCRIPTION
**Pull request by pmishchenko-ua**

### CODE REVIEW

File: ma_dsn.c [hunk 1]
```diff
@@ -86,6 +86,7 @@ MADB_DsnKey DsnKeys[]=
   {"APP",            offsetof(MADB_Dsn, App),               DSN_TYPE_STRING, 0, 0},
   /* SSO parameters */
   {"BROWSER_SSO",    offsetof(MADB_Dsn, IsBrowserAuth),     DSN_TYPE_BOOL  , 0, 0},
+  {"NO_KEYRING_SSO", offsetof(MADB_Dsn, HasNoKeyring),      DSN_TYPE_BOOL  , 0, 0},
```
Comment (AdalbertMemSQL) [inline ma_dsn.c:89]: nit:
What about `IGNORE_KEYRING_SSO` and `IgnoreKeyring` or smth like this?

File: plugins/browser_auth.c [hunk 2]
```diff
@@ -817,19 +814,18 @@ MDBUG_C_RETURN(Dbc, Dbc->Error.ReturnValue, &Dbc->Error);
   GError *error = NULL;
   secret_password_store_sync(JWT_CACHE_SCHEMA, SECRET_COLLECTION_DEFAULT,
                             SECURE_JWT_STORAGE_KEY, bac->token, NULL, &error,
-                            "email", bac->email,
                             NULL);
   if (error != NULL)
   {
     MDBUG_C_PRINT(Dbc, "%s FAILED", "secret_password_store_sync");
-    MADB_SetError(&Dbc->Error, error->code, error->message, 0);
+    MADB_SetError(&Dbc->Error, MADB_ERR_HY000, error->message, 0);
```
Comment (AdalbertMemSQL) [inline plugins/browser_auth.c:821]: I don't think that error returned by secret_password_store_sync matches the ODBC error codes

Comment (pmishchenko-ua) [reply plugins/browser_auth.c:821]: you're right, that's why the previous version produced a wrong result (see https://memsql.atlassian.net/browse/PLAT-6231)

Comment (AdalbertMemSQL) [reply plugins/browser_auth.c:821]: Ou, I see.
I don't get used to the GitHub UI and thought that it was changed in the other way :)

File: ma_dsn.c (hunk 1 above)
Comment (pmishchenko-ua) [reply ma_dsn.c]: fine with me

File: ma_connection.c [hunk 3]
```diff
@@ -590,6 +590,18 @@ SQLRETURN MADB_DbcConnectDB(MADB_Dbc *Connection,
   if (!Connection || !Dsn)
     return SQL_ERROR;
 
+
+  char *emailForSSO = Dsn->UserName;
+  my_bool connectionTried = FALSE;
```
[the remark this answers is not in the document]
Comment (pmishchenko-ua) [reply ma_connection.c:595]: it is used on line 825

[the remark this answers is not in the document]
Comment (AdalbertMemSQL) [reply ma_connection.c:595]: Hmm... 
Can you move it before `real_connect:` and leave a good comment?

File: plugins/browser_auth.c [hunk 4]
```diff
@@ -642,7 +641,9 @@ int GetCachedCredentials(MADB_Dbc *Dbc, const char *email, BrowserAuthCredential
 // Mac secure key storage
  void *password_data = NULL;
  uint32_t password_len;
- char *email_str = email != NULL ? email : "";
+ // char *email_str = email != NULL ? email : "";
+ // We don't use email for now as we don't support multiple users
+ char *email_str = "";
```
[the remark this answers is not in the document]
Comment (AdalbertMemSQL) [reply plugins/browser_auth.c:646]: Got it.
Then it LGTM.

File: ma_connection.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (pmishchenko-ua) [reply ma_connection.c]: sure, added a comment

**Review by AdalbertMemSQL — APPROVED** [review]

**Pull request merged**